pkg/kubelet/cm/dra/claiminfo.go

```diff
@@ -118,10 +148,25 @@ func (cache *claimInfoCache) hasPodReference(UID types.UID) bool {
 	defer cache.RUnlock()
 
 	for _, claimInfo := range cache.claimInfo {
-		if claimInfo.podUIDs.Has(string(UID)) {
+		if claimInfo.PodUIDs.Has(string(UID)) {
 			return true
 		}
 	}
 
 	return false
 }
+
+func (cache *claimInfoCache) syncToCheckpoint() {
+	cache.RLock()
+	defer cache.RUnlock()
+
+	claimInfoStateList := make(state.ClaimInfoStateList, 0)
+	for _, infoClaim := range cache.claimInfo {
+		claimInfoStateList = append(claimInfoStateList, infoClaim.ClaimInfoState)
+	}
```

Comment:
Why we should convert `claimInfo` map to the list?

Reply:
because we don't want to relay on the calimInfo cache key. Currently it claimName+namespace in the future it can be something else. if we checkpoint it and change it in the future we will need to migrate from old format to new format of the key.

Reply:
This method assumes that we have enough info in the checkpoint to reconstruct the cache from it. In this particular case it means that we must always have a claim name and a namespace in the checkpoint data. This assumption is a bit concerning.

@moshe010
> Currently it claimName+namespace in the future it can be something else.

true. However, in this case you'll have to add this something else to the checkpoint data, which would be a format change anyway. Otherwise you won't be able to restore the cache from the checkpoint.

Reply:
The other way to look at it is that we are simply checkpointing the full state of the claim necessary to:
1. make future `NodeUnprepareResource()` calls on already prepared resources
2. cache any information necessary for exposure through the PodResources API

This is exactly what we store in the checkpoint -- no more, no less. What do you find concerning about this?

Reply:
> What do you find concerning about this?

The fact that we store only part of the cache and changing its structure. I find it a bit concerning, but don't feel strongly about it.

Reply:
I don't share your concern. Only the bits absolutely necessary for checkpointing are synced to disk and everything else that can be generated from that (for convenience) is done so as the cache is repopulated after a kubelet restart.

Reply:
What's the purpose of this division?

Reply:
Less churn in the checkpoint format (since it only stores the essentials) and the ability to easily add meta-fields as we see fit hanging off of the claimInfo struct.

Reply:
I'm generally ok with this. It's more complex than storing the whole cache though. I'm not sure it makes much sense currently as we only generate annotations, but it can make sense in future if we have more metadata, especially if that metadata is hard or time consuming to generate.